File: app/src/main/java/com/cornellappdev/android/pollo/Models/Poll.java
package com.cornellappdev.android.pollo.models;

import org.json.JSONObject;

public class Poll {

    private Long id;
    private String text;
    private JSONObject results;
    private boolean shared;
    private String type;
    private String answer;

    public Poll(Long id, String text, JSONObject results, boolean shared, String type, String answer) {
        this.id = id;
        this.text = text;
        this.results = results;
        this.shared = shared;
        this.type = type;
        this.answer = answer;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public JSONObject getResults() {
        return results;
    }

    public void setResults(JSONObject results) {
        this.results = results;
    }

    public boolean isShared() {
        return shared;
    }

    public void setShared(boolean shared) {
        this.shared = shared;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getAnswer() {
        return answer;
    }

    public void setAnswer(String answer) {
        this.answer = answer;
    }
}

File: app/src/main/java/com/cornellappdev/android/pollo/models/Poll.kt
package com.cornellappdev.android.pollo.models

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class PollResult(val index: Int, val text: String, val count: Int?): Parcelable

@Parcelize
enum class PollState: Parcelable {
    live, ended, shared
}

@Parcelize
data class Poll(val createdAt: String?, val updatedAt: String?, var id: String?,
                val text: String, val answerChoices: ArrayList<PollResult>,
                val correctAnswer: Int, val userAnswers: MutableMap<String, ArrayList<Int>>,
                val state: PollState): Parcelable